frontend/src/components/workspaces/Templates/TemplatesTableLogic/TemplatesTableLogic.tsx

```diff
@@ -0,0 +1,118 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Spin } from 'antd';
+import { Dispatch, SetStateAction, useContext, useEffect } from 'react';
+import { FC } from 'react';
+import { AuthContext } from '../../../../contexts/AuthContext';
+import {
+  useOwnedInstancesQuery,
+  useWorkspaceTemplatesQuery,
+} from '../../../../generated-types';
+import { VmStatus, WorkspaceRole } from '../../../../utils';
+import { TemplatesEmpty } from '../TemplatesEmpty';
+import { TemplatesTable } from '../TemplatesTable';
+
+export interface ITemplateTableLogicProps {
+  workspaceNamespace: string;
+  role: WorkspaceRole;
+  reload: boolean;
+  setReload: Dispatch<SetStateAction<boolean>>;
+}
+
+const TemplatesTableLogic: FC<ITemplateTableLogicProps> = ({ ...props }) => {
+  const { userId } = useContext(AuthContext);
+  const { workspaceNamespace, role, reload, setReload } = props;
+
+  const {
+    data: dataTemplate,
+    loading: loadingTemplate,
+    error: errorTemplate,
+    refetch: refetchTemplate,
+  } = useWorkspaceTemplatesQuery({
+    variables: { workspaceNamespace: workspaceNamespace },
+    notifyOnNetworkStatusChange: true,
+  });
+
+  const {
+    data: dataInstances,
+    loading: loadingInstances,
+    error: errorInstances,
+    startPolling: startPollingInstances,
+  } = useOwnedInstancesQuery({
+    variables: { tenantNamespace: `tenant-${userId}` },
+  });
+
+  startPollingInstances(500);
+
+  useEffect(() => {
```

Comment:
If this is not temporary (maybe you are waiting for the graphql watchers)
Add some comments to this file to explain all the logic regarding `reload`, `refetchTemplate`, etc